### ISSUE TITLE
Fails on `dofle#123`

### ISSUE THREAD

**Issue by borekb** [sindresorhus/issue-regex#1]
`dofle#123` is not a valid issue reference but this package matches it. I'll try to work out the correct regexp.

**Comment by borekb** [comment]
This would work if lookbehinds were available:

`([\w-.]+\/[\w-.]+#[1-9]\d*|(?<=\s)#[1-9]\d*)`

Not yet sure how to emulate it in JavaScript. [Playground on Regex101](https://regex101.com/r/SQrOlx/2).

**Comment by borekb** [comment]
This regex will contain the correct match in group 1:

```
(?:[^\S\r\n]|^)((?:[\w-.]+\/[\w-.]+)?#[1-9]\d*)
```

[Regex101](https://regex101.com/r/SQrOlx/4)

However, that would break the interface of this package, for example, it would be impossible to do:

```js
'Fixes #143 and avajs/ava#1023'.match(require('issue-regex'));
```

I'm not sure there is a solution that would both work correctly and match without groups.

**Comment by sindresorhus** [comment]
Lookbehinds are in latest Chrome, so maybe we could feature detect it and at least fix it for the latest Chrome and upcoming Node.js versions: https://github.com/kevva/url-regex/pull/35#issuecomment-331970744 ?

**Comment by borekb** [comment]
Good idea. What would you think about exporting not only the regexp itself but also a function that would provide matches similar to `'string'.match()`?

**Comment by borekb** [comment]
[Improved regex](https://regex101.com/r/SQrOlx/5).

**Comment by borekb** [comment]
The correct regex is `(?:[\w-.]+\/[\w-.]+|\B)#[1-9]\d*`, the fix is in #2.

**Comment by sindresorhus** [comment]
> What would you think about exporting not only the regexp itself but also a function that would provide matches similar to 'string'.match()?

Yeah, I think that would be useful. One method to return an array of issues in a string and one method to test if a string is a valid issue reference.
